caffe2/serialize/inline_container.h

```diff
@@ -119,135 +134,202 @@ class PyTorchStreamReader {
+    cursor_ += size;
+    seekToNextAlignmentBoundary();
+    return std::tuple<at::DataPtr, int64_t, int64_t>(
+        std::move(retval), key, size);
+  }
+
+  bool hasNextRecord() {
+    if (cursor_ + kFieldAlignment * 2 < file_size_) {
+      return true;
+    }
+    return false;
+  }
+
+  bool close() {
+    if (closed_) {
+      return false;
+    }
+    closed_ = true;
+    return true;
+  }
+
+  bool closed() {
+    return closed_;
+  }
+
   ~PyTorchStreamReader() {
   }
+
  private:
-  std::istream& in;
-  size_t cursor = 0;
-  size_t file_size;
-  size_t last_record_offset;
+  std::istream& in_;
+  size_t cursor_ = 0;
+  size_t file_size_;
+  size_t last_record_offset_;
+  bool closed_ = false;
 
   // Utility functions
   uint64_t read64BitIntegerLittleEndian() {
-   uint64_t retval;
-   // TODO endian swap on platforms that need it?
-   in.read(reinterpret_cast<char *>(&retval), 8);
-   std::streamsize read_bytes = in.gcount();
-   if (read_bytes != 8) {
-     std::ostringstream errmsg;
-     errmsg << "Expected to read 8 bytes but got " << read_bytes;
-     throw std::runtime_error(errmsg.str());
-   }
-   cursor += read_bytes;
-   return retval;
+    uint64_t retval;
+    // TODO endian swap on platforms that need it?
+    in_.read(reinterpret_cast<char*>(&retval), 8);
+    std::streamsize read_bytes = in_.gcount();
+    if (read_bytes != 8) {
+      std::ostringstream errmsg;
+      errmsg << "Expected to read 8 bytes but got " << read_bytes;
+      throw std::runtime_error(errmsg.str());
+    }
+    cursor_ += read_bytes;
+    return retval;
   }
 
   void seekToNextAlignmentBoundary() {
-   size_t next_offset = (cursor + kFieldAlignment) - (cursor % kFieldAlignment);
-   size_t pad_amount = next_offset - cursor;
-   cursor += pad_amount;
-   in.seekg(cursor);
+    size_t next_offset =
+        (cursor_ + kFieldAlignment) - (cursor_ % kFieldAlignment);
+    size_t pad_amount = next_offset - cursor_;
+    cursor_ += pad_amount;
+    in_.seekg(cursor_);
   }
 
   // File format deserialization functions
-  void readAndValidateFileHeader() {
-   // Validate magic number
-   uint64_t magic = read64BitIntegerLittleEndian();
-   if (magic != kFileMagicNumber) {
-     throw std::runtime_error("Magic number mismatch in PyTorch file. File may"
-                              " be corrupted or is not actually a PyTorch file.");
-   }
-   uint64_t file_format_version = read64BitIntegerLittleEndian();
-   if (file_format_version > kMaxSupportedFileFormatVersion) {
-     std::ostringstream errmsg;
-     errmsg << "Attempted to read a PyTorch file with version " << file_format_version
-            << " but the maximum supported version for reading is " << kMaxSupportedFileFormatVersion
-            << ". Your PyTorch installation may be too old.";
-     throw std::runtime_error(errmsg.str());
-   }
-   seekToNextAlignmentBoundary();
-  }
-  void readAndValidateFileFooter() {
+  bool readAndValidateFileHeader() {
```

Comment:
I'm not sure if it's a good idea to throw exceptions in this function. The return value is a bool while besides throwing exceptions, it will always return true here.

Reply:
Yeah, use LOG, and return false instead :-)